int32 x
int32 y
uint8[] descriptor
int32 feature_index
